### DISCUSSION TITLE
Support needed for Japanese players who wish to mod.

### DISCUSSION THREAD

**Discussion by zaphod77** [tsunamods-codes/7th-Heaven#232] [Ideas]
Japanese players have only two options to purchase the game on pc.

1) Square Enix store version.  They _cannot purchase it on steam_, and it is different from the usual release, as it has some new cheat hooks in it's driver.
2) Windows Edition from the ms store, which is freely available and allows language changing. Uses a UWP wrapper to run it's exe, and has two of them. the english exe, for all versions except japanese, and a ff7_jp exe which exists nowhere else except maybe the e-store japan release.

While i can't be of much help for the jp e-store version, which requires a credit card with a japanese billing address to buy, i do have the windows store version, which is the same everywhere it is bought.

I can provide info on detecting an installed windows store edition, but it cannot be supported natively, because of the uwp wrapper.  This means it can only be supported by conversion. a similar process to the one that i s used for converting steam editions should work, and if i'm contacted i'm willing to help work this out. Thankfully, there is no crack for windows store edition 1.0.2.0.  It's also possible to directly check using c sharp if any UWP app is installed.

**Comment by julianxhokaxhiu** [comment]
Thanks for opening up this discussion.

Regarding the eStore version, I am already working with someone in Japan willing to support and understand what is not currently working ( however 7th code is already wired to support that edition, so might be a bug or left over implementation somehow ).

About the Windows Edition from Microsoft Store, we are not interested in converting the game. 7th will no more downgrade game editions, it will use them as they are. Because I do not own a copy of the game it's hard to spot how we can make it work, but if you're willing to work on a converter yourself feel free. However we are not interested in those because of legal reasons, so feel free to use it at your own risk.

Hope this helps, cheers!

**Comment by zaphod77** [comment]
The main reason i can think of for the JP e-store edition to not work is there not being a ff7_en.exe, just a ff7_jp.exe.  if there is an en executable, it should work, goinf by my reading of the code.  There are plenty of mods that should be possible to install into a JP version with JP text, though, and it would be nice for them to have those.

My thought was that verification of ownership of the windows edition was doable enough that such an approach can be considered, but i understand your reluctance to do it. i'm not good enough to make the converter myself, an di already have a steam version too.  But the windows store version has FULL contorller remapping support. as when you change the mapping in game, the prompts are fixed too. it even says "enter" instead of "start" when you are playing on keyboard!  This seems to be a feature that could be implemented inside of FFNX.
